### PR TITLE
feat(eks): eks fault injection using k8s custom resource

### PR DESCRIPTION
**Pull request by Young-ook**
*Issue #, if available:*

*Description of changes:* add new eks fault injection experiment template with kubernetes custom resource.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: eks/cdk/eks.py [hunk 1]
```diff
@@ -64,7 +66,7 @@ def __init__(self, app: core.App, id: str, props, **kwargs) -> None:
 
         eks.add_helm_chart("cluster-autoscaler",
             chart = "cluster-autoscaler",
-            release = "cluster-autoscaler",
+            release = "aws-cluster-autoscaler",
```
Comment (Young-ook) [inline eks/cdk/eks.py:69]: Change the name of the Helm chart release to be the same as the kubernetes deployment

File: eks/cdk/eks.py [hunk 2]
```diff
@@ -75,14 +77,38 @@ def __init__(self, app: core.App, id: str, props, **kwargs) -> None:
...
+        fis_role.add_managed_policy(aws_iam.ManagedPolicy.from_aws_managed_policy_name("AdministratorAccess"))
+
+        with open('../kubernetes/manifest/cm-manager.yaml', 'r') as f:
+            cmm_manifest = yaml.safe_load(f)
+
+        eks.add_manifest("cm-manager", cmm_manifest)
+        f.close()
+
+        # update aws-auth config map
+        eks.aws_auth.add_role_mapping(fis_role, groups=["system:masters", "chaos-mesh-manager-role"])
```
Comment (Young-ook) [inline eks/cdk/eks.py:100]: This is code to create a kuberentes configmap in the kube-system namespace for IAM roles and Kuberentes RBAC mappings.

File: eks/cdk/fis.py [hunk 3]
```diff
@@ -12,10 +13,6 @@ def __init__(self, app: core.App, id: str, props, **kwargs) -> None:
         # copy properties
         self.output_props = props.copy()
 
-        # iam role
-        role = aws_iam.Role(self, "fis-role", assumed_by = aws_iam.ServicePrincipal("fis.amazonaws.com"))
-        role.add_managed_policy(aws_iam.ManagedPolicy.from_aws_managed_policy_name("AdministratorAccess"))
```
Comment (Young-ook) [inline eks/cdk/fis.py:17]: The FiS role creation code has been moved into the EKS stack to avoid circular dependencies.

File: eks/cdk/fis.py [hunk 4]
```diff
@@ -110,6 +107,42 @@ def __init__(self, app: core.App, id: str, props, **kwargs) -> None:
...
+                resource_arns = [f"{props['eks'].cluster_arn}"],
+                selection_mode = "ALL"
+            )},
+            actions = {'eks-kill-pod': aws_fis.CfnExperimentTemplate.ExperimentTemplateActionProperty(
+                action_id = "aws:eks:inject-kubernetes-custom-resource",
+                parameters = dict(
+                    kubernetesApiVersion = "chaos-mesh.org/v1alpha1",
+                    kubernetesKind = "PodChaos",
+                    kubernetesNamespace = "chaos-mesh",
+                    kubernetesSpec = json.dumps({
```
Comment (Young-ook) [inline eks/cdk/fis.py:126]: This part can be changed with your manifests if you want to inject different fault scenarios into the kubernetes objects. For more details, please refer to https://chaos-mesh.org/docs/run-a-chaos-experiment/

**Comment by cjk0604** [comment]
Noted

**Pull request merged**